lib/puppet_litmus/puppet_helpers.rb

```diff
@@ -303,6 +303,64 @@ def run_bolt_task(task_name, params = {}, opts = {})
     end
   end
 
+  # Runs a plan against the target system.
+  #
+  # @param plan_name [String] The name of the task to run.
+  # @param params [Hash] key : value pairs to be passed to the task.
+  # @param opts [Hash] Alters the behaviour of the command. Valid options are
+  #  :expect_failures [Boolean] doesnt return an exit code of non-zero if the command failed.
+  #  :inventory_file [String] path to the inventory file to use with the task.
+  # @return [Object] A result object from the task.The values available are stdout, stderr and result.
+  def run_bolt_plan(plan_name, params = {}, opts = {})
+    Honeycomb.start_span(name: 'litmus.run_plan') do |span|
+      ENV['HTTP_X_HONEYCOMB_TRACE'] = span.to_trace_header
+      span.add_field('litmus.plan_name', plan_name)
+      span.add_field('litmus.params', params)
+      span.add_field('litmus.opts', opts)
+
+      config_data = { 'modulepath' => File.join(Dir.pwd, 'spec', 'fixtures', 'modules') }
+      target_node_name = targeting_localhost? ? 'litmus_localhost' : ENV['TARGET_HOST']
+      inventory_hash = if !opts[:inventory_file].nil? && File.exist?(opts[:inventory_file])
+                         inventory_hash_from_inventory_file(opts[:inventory_file])
+                       elsif File.exist?('inventory.yaml')
+                         inventory_hash_from_inventory_file('inventory.yaml')
+                       else
+                         localhost_inventory_hash
+                       end
+      raise "Target '#{target_node_name}' not found in inventory.yaml" unless target_in_inventory?(inventory_hash, target_node_name)
+
+      span.add_field('litmus.node_name', target_node_name)
+      add_platform_field(inventory_hash, target_node_name)
+
+      bolt_result = run_plan(plan_name, params, config: config_data, inventory: inventory_hash)
+      # Plans don't provide stdin and stdout streams. They only have a result.
+      # They don't provide an exit code either, just a 'success' value.
+      result_obj = {
+        status: nil,
+        result: nil,
+      }
```

Comment:
```
result_obj = {
        exit_code: 0,
        stdout: nil,
        stderr: nil,
        result: bolt_result.first['value'],
      }
```
The reason for this is to allow backwards compatibility for beaker style tests. 
Granted we are adding something that beaker never had, but people still write tests where they check for status code, even though there is no status code. they just want to know if it failed with a .zero? check
You should be able to copy from https://github.com/puppetlabs/puppet_litmus/blob/b4f659013855d5900f2481bd2cc9fc2f56b7af2f/lib/puppet_litmus/puppet_helpers.rb#L245

Reply:
Yeah I can add those. I'll just a comment in the code that they aren't real. Just placeholders as you say for compatibility.

Reply:
Can I get away with just adding the exit code and not sdtout and stderr?

Reply:
Don't mind me. I decided to add those in too.